mongoimport/mongoimport.go

```diff
@@ -154,17 +161,30 @@ func (imp *MongoImport) ValidateSettings(args []string) error {
 		}
 	}
 
+	if imp.IngestOptions.Upsert == true {
+		imp.IngestOptions.Mode = MODE_UPSERT
+	}
+	if imp.IngestOptions.Mode == "" {
+		imp.IngestOptions.Mode = MODE_INSERT
+	} else if !(imp.IngestOptions.Mode == MODE_INSERT ||
+		imp.IngestOptions.Mode == MODE_UPSERT ||
+		imp.IngestOptions.Mode == MODE_MERGE) {
+		return fmt.Errorf("invalid --mode argument: %v", imp.IngestOptions.Mode)
+	}
+
 	if imp.IngestOptions.UpsertFields != "" {
-		imp.IngestOptions.Upsert = true
+		if imp.IngestOptions.Mode == MODE_INSERT {
+			imp.IngestOptions.Mode = MODE_UPSERT
```

Comment:
Ok, fair enough. 

The problem with the code as it is is that if you were to specify --upsertFields "foo" --mode=insert, then you'd silently ignore their explicit request to do inserts.

I think what you need to do is to remove the default value for the mode option, so that you can detect the difference between not specifying the option, and specifying "insert"